### ISSUE TITLE
JPEG XL rendering

### ISSUE THREAD

**Issue by H-A-M-G-E-R** [Knewest/embed-more-images#2]
maybe convert JPEG XL to PNG using javascript and then render it?

**Comment by Knewest** [comment]
Hi, thanks for your interest in my projects. I am just confirming with you that this is planned, among other formats. No promises, though.

**Comment by goodusername123** [comment]
Discord currently can natively render/use JXL images if you pass `-enable-features=JXL` as a launch parameter when opening it, This will continue to work until Discord drops support for Windows 7/8/8.1 since the version of Chromium that deleted all the JXL code is the same one that dropped support for anything older then Windows 10. In the meantime could JXL images be allowed to natively render in this plugin until another solution is implemented?

Maybe possibly have a toggle or something for it that says/notes somewhere that it requires `-enable-features=JXL` and will stop working in versions of the Discord client after support for Windows 7/8/8.1 is dropped? (unsure when Discord will decide to do this).

**Comment by jonnyawsom3** [comment]
Stumbled across this, so thought I'd share the JXL-Oxide WASM decoder, which is bundled in the github releases.
https://jxl-oxide.tirr.dev/demo/index.html
https://github.com/tirr-c/jxl-oxide

It's been ported to a browser extension before, but unfortunately the repo was deleted.

**Comment by soup587** [comment]
chromium's supposed to have jpegxl support behind a flag and have it be stable sometime september, can this be revisited?

**Comment by Knewest** [comment]
> chromium's supposed to have jpegxl support behind a flag and have it be stable sometime september, can this be revisited?

Yes. I will revisit this properly during my current rewrite of this plugin.
